includes/KSES.php

```diff
@@ -74,7 +86,55 @@ public function register(): void {
 	 * @return string[] List of required services.
 	 */
 	public static function get_requirements(): array {
-		return [ 'story_post_type' ];
+		return [ 'story_post_type', 'page_template_post_type' ];
+	}
+
+	/**
+	 * Checks whether the post type is correct and user has capability to edit it.
+	 *
+	 * @since 1.22.0
+	 *
+	 * @param string   $post_type   Post type slug.
+	 * @param int|null $post_parent Parent post ID.
+	 * @return bool Whether the user can edit the provided post type.
+	 */
+	private function is_allowed_post_type( string $post_type, ?int $post_parent ): bool {
+		if ( $this->story_post_type->get_slug() === $post_type && $this->story_post_type->has_cap( 'edit_posts' ) ) {
+			return true;
+		}
+
+		if ( $this->page_template_post_type->get_slug() === $post_type && $this->page_template_post_type->has_cap( 'edit_posts' ) ) {
+			return true;
+		}
+
+		// For story autosaves.
+		if (
+			(
+				'revision' === $post_type &&
+				! empty( $post_parent ) &&
+				get_post_type( $post_parent ) === $this->story_post_type->get_slug()
+			) &&
+			$this->story_post_type->has_cap( 'edit_posts' )
+		) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Filters story data.
+	 *
+	 * Provides simple sanity check to ensure story data is valid JSON.
+	 *
+	 * @since 1.22.0
+	 *
+	 * @param string $story_data JSON-encoded story data.
+	 * @return string Sanitized & slashed story data.
+	 */
+	private function filter_story_data( string $story_data ): string {
+		$decoded = json_decode( (string) wp_unslash( $story_data ), true );
+		return null === $decoded ? '' : wp_slash( (string) wp_json_encode( $decoded ) );
```

Comment:
This feels like witchcraft atm. Can you provide a unit test with invalid data to proof this works as expected.